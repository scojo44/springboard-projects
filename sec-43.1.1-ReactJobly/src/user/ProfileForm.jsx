import React, {useContext} from 'react'
import {useForm} from 'react-hook-form'
import UserContext from '../UserContext'
// import './ProfileForm.css'

export default function ProfileForm({update}) {
  const {currentUser} = useContext(UserContext);
  const {register, handleSubmit, formState: {errors}} = useForm({
    defaultValues: {
      username: currentUser.username,
      firstName: currentUser.firstName,
      lastName: currentUser.lastName,
      email: currentUser.email
    }
  });

  return (
    <form className="ProfileForm" onSubmit={handleSubmit(async credentials => await update(credentials))}>
      <h2>Edit Profile</h2>
      <p>
        <label htmlFor="username">Username: </label>
        <input readOnly {...register("username")} />
      </p>
      <p>
        <label htmlFor="password">New Password (leave blank to not change): </label>
        <input type="password" autoComplete="new-password" {...register("password")} />
        {errors.password && <span className="input-error"> {errors.password.message}</span>}
      </p>
      <p>
        <label htmlFor="confirm">Confirm New Password: </label>
        <input type="password" autoComplete="new-password"
          {...register("confirm", {validate: (cp, values) => cp === values.password || 'Passwords do not match'})}
        />
        {errors.confirm && <span className="input-error"> {errors.confirm.message}</span>}
      </p>
      <p>
        <label htmlFor="firstName">First Name: </label>
        <input {...register("firstName", {required: 'Please enter your first name'})} />
        {errors.firstName && <span className="input-error"> {errors.firstName.message}</span>}
      </p>
      <p>
        <label htmlFor="lastName">Last Name: </label>
        <input {...register("lastName", {required: 'Please enter your last name'})} />
        {errors.lastName && <span className="input-error"> {errors.lastName.message}</span>}
      </p>
      <p>
        <label htmlFor="email">Email: </label>
        <input type="email" {...register("email", {required: 'Please enter your email address', pattern: /.+@[a-z0-9-]+/i})} />
        {errors.email && <span className="input-error"> {errors.email.message}</span>}
      </p>
      <button type="submit">Save</button>
    </form>
  );
}
