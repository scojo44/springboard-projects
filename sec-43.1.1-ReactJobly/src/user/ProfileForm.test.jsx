import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import UserContext from '../UserContext'
import ProfileForm from './ProfileForm'

const context = {
  currentUser: {
    username: 'testuser',
    firstName: 'First',
    lastname: 'Last',
    email: 'first@last.test'
  }
};

describe('ProfileForm Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <ProfileForm />
        </UserContext.Provider>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <ProfileForm />
        </UserContext.Provider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
