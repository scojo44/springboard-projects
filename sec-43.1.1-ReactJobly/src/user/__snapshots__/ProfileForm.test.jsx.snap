// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileForm Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ProfileForm"
  >
    <h2>
      Edit Profile
    </h2>
    <p>
      <label
        for="username"
      >
        Username: 
      </label>
      <input
        name="username"
        readonly=""
      />
    </p>
    <p>
      <label
        for="password"
      >
        New Password (leave blank to not change): 
      </label>
      <input
        autocomplete="new-password"
        name="password"
        type="password"
      />
    </p>
    <p>
      <label
        for="confirm"
      >
        Confirm New Password: 
      </label>
      <input
        autocomplete="new-password"
        name="confirm"
        type="password"
      />
    </p>
    <p>
      <label
        for="firstName"
      >
        First Name: 
      </label>
      <input
        name="firstName"
      />
    </p>
    <p>
      <label
        for="lastName"
      >
        Last Name: 
      </label>
      <input
        name="lastName"
      />
    </p>
    <p>
      <label
        for="email"
      >
        Email: 
      </label>
      <input
        name="email"
        type="email"
      />
    </p>
    <button
      type="submit"
    >
      Save
    </button>
  </form>
</DocumentFragment>
`;
